src/modules/bootloader/main.py

```diff
@@ -605,17 +605,71 @@ def run_grub_install(fw_type, partitions, efi_directory):
                                    "--target=" + efi_target,
                                    "--efi-directory=" + efi_directory,
+                                   '--modules="' + grub_modules + '"',
+                                   "--disable-shim-lock",
+                                   "--sbat /usr/share/grub/sbat.csv",
                                    "--bootloader-id=" + efi_bootloader_id,
-                                   "--force"])
+                                   "--force"])])
+        MOK_path = "/etc/secureboot/"
+        if os.path.exists(install_path + MOK_path + "MOK.key"):
+            grub_path = efi_directory + "/EFI/" + efi_bootloader_id + "/" + efi_grub_file
+            check_target_env_call(["sh", "-c", "/usr/bin/find /boot/ -maxdepth 1 -name 'vmlinuz-*' | /usr/bin/xargs -I{} /bin/sh -c 'if ! /usr/bin/sbverify --cert /etc/secureboot/MOK.crt {} >/dev/null 2>&1 ;then sudo /usr/bin/sbsign --key " + MOK_path + "MOK.key --cert " + MOK_path + "MOK.crt --output {} {}; fi'"])
+            check_target_env_call(["sh", "-c", "/usr/bin/sudo /usr/bin/sbsign --key " + MOK_path + "MOK.key --cert " + MOK_path + "MOK.crt --output " + grub_path + " " + grub_path])
+            check_target_env_call(["sh", "-c", "/usr/bin/sudo /usr/bin/mokutil --import " + MOK_path + "MOK.cer --root-pw"])
+            install_efi_directory = install_path + efi_directory
+            shutil.copy2(os.path.join(install_path, "/usr/share/shim-signed/shimx64.efi") , os.path.join(install_efi_directory, "EFI", efi_bootloader_id))
+            shutil.copy2(os.path.join(install_path, "/usr/share/shim-signed/mmx64.efi") , os.path.join(install_efi_directory, "EFI", efi_bootloader_id))
+            shutil.copy2(os.path.join(install_path, "/usr/share/shim-signed/shimx64.efi") , os.path.join(install_efi_directory, "EFI", efi_bootloader_id, "bootx64.efi"))
+            efi_drive = subprocess.check_output([
+            libcalamares.job.configuration["grubProbe"],
+            "-t", "drive", install_efi_directory]).decode("ascii")
+            efi_disk = subprocess.check_output([
+            libcalamares.job.configuration["grubProbe"],
+            "-t", "disk", install_efi_directory]).decode("ascii")
+            efi_drive_partition = efi_drive.replace("(","").replace(")","").split('/')[3]
+            # Get the first run of digits from the partition
+            efi_partition_number = None
+            c = 0
+            start = None
+            while c < len(efi_drive_partition):
+                if efi_drive_partition[c].isdigit() and start is None:
+                    start = c
+                if not efi_drive_partition[c].isdigit() and start is not None:
+                    efi_partition_number = efi_drive_partition[start:c]
+                    break
+                c += 1
+            if efi_partition_number is None:
+                raise ValueError("No partition number found for %s" % install_efi_directory)
```

Comment:
Don't raise an exception unless you are going to handle it somewhere.

In this case, I think it would make more sense to use `libcalamares.util.error()`